### PR TITLE
Only give same values .selected class in gr.Dropdown

### PR DESCRIPTION
**Pull request by space-nuko**
# Description

Closes: #3564

Still gives elements with the same value in the same dropdown the selected checkmark, unsure if this is desirable

2023-03-27 16_32_01-Gradio - Chromium

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have added a short summary of my change to the CHANGELOG.md
- [x] My code follows the style guidelines of this project
- [x] I have commented my code in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: js/form/src/DropdownOptions.svelte [hunk 1]
```diff
@@ -49,7 +49,7 @@
 			<li
 				class="item"
 				role="button"
-				class:selected={value?.includes(choice)}
+				class:selected={value === choice}
```
Comment (aliabid94) [inline js/form/src/DropdownOptions.svelte:52]:
```suggestion
				class:selected={Array.isArray(value) ? value.includes(choice) : value === choice}
```
So value can either be a single string, or an array of strings in multiselect mode. I believe this is why the original logic used "includes" to handle multiselect mode, but this suggestion should handle both cases.

**Review by pngwn — APPROVED** [review]
When we've fixed the changelog and run `pnpm format:write` to fix the formatting, i think this is good to go.

File: js/form/src/DropdownOptions.svelte [hunk 2]
```diff
@@ -32,6 +32,11 @@
 	}
 
 	const dispatch = createEventDispatcher();
+
+	function isSelected(choice) {
```
Comment (pngwn) [inline js/form/src/DropdownOptions.svelte:36]:
```suggestion
	function isSelected(choice: string) {
```

**Review by pngwn — CHANGES_REQUESTED** [review]
This should fix the TS error.

**Review by pngwn — APPROVED** [review]

**Pull request merged**